product-service/src/getProductsById.ts

```diff
@@ -0,0 +1,67 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import 'source-map-support/register';
+import productList from "../productList.json";
+import axios from 'axios'
+
+// function is for async/awit try/catch example
+const getWeatherInfo = async (productId) => {
+  
+  const OPEN_WEATHER_URL = 'http://api.openweathermap.org/data/2.5/weather'
+  let apiKey = '2b5437d79044ea0ae5c424019c101372'
+  let cityName = productId || 'Barnaul'
+
+  if(productId === 'test_catch'){
+    apiKey = 'mock_unactive_api_key'
+  }
+
+  try{
+    const { data } =  await axios.get(`${OPEN_WEATHER_URL}?q=${cityName}&appid=${apiKey}`) 
+    return data;
+  } catch (error) {
+    return {
+      message: 'Unable to get weather info'
+    }
+  }
+}
+
+export const getProductsById: APIGatewayProxyHandler = async (event, _context) => {
+  
+  let statusCode = 404;
+  let result = {  }
+  
+  try {
+    
+    const { pathParameters:{productId}} = event;
+    
+    //  needed as awit using example
+    const  weatherData = await getWeatherInfo(productId)
+
+    const filteredProducts = productList.filter((product)=> product.id === productId)
```

Comment:
You can use productsList.find... that will return you product

Reply:
I've replaced filter with find.